climada/entity/impact_funcs/trop_cyclone.py

```diff
@@ -311,118 +441,74 @@ def get_countries_per_region(region=None):
-                "ESP", "SDN", "SJM", "SWE", "CHE", "TGO", "TUN", "TUR", "UKR", "GBR",
-                "UMI", "ESH", "ZMB", "ALA",
-            ],
-        }
-        # fmt: on
-        impf_id = {
-            "NA1": 1,
-            "NA2": 2,
-            "NI": 3,
-            "OC": 4,
-            "SI": 5,
-            "WP1": 6,
-            "WP2": 7,
-            "WP3": 8,
-            "WP4": 9,
-            "ROW": 10,
-        }
-        region_name = dict()
-        region_name["NA1"] = "Caribbean and Mexico"
-        region_name["NA2"] = "USA and Canada"
-        region_name["NI"] = "North Indian"
-        region_name["OC"] = "Oceania"
-        region_name["SI"] = "South Indian"
-        region_name["WP1"] = "South East Asia"
-        region_name["WP2"] = "Philippines"
-        region_name["WP3"] = "China Mainland"
-        region_name["WP4"] = "North West Pacific"
 
         if region == "all":
-            return region_name, impf_id, iso3n, iso3a
+            return (
+                CountryCode.REGION_NAME.value,
+                CountryCode.IMPF_ID.value,
+                CountryCode.ISO3N.value,
+                CountryCode.ISO3A.value,
+            )
+
+        return (
+            CountryCode.REGION_NAME.value[region],
+            CountryCode.IMPF_ID.value[region],
+            CountryCode.ISO3N.value[region],
+            CountryCode.ISO3A.value[region],
+        )
+
+    @staticmethod
+    def get_impf_id_regions_per_countries(
+        countries: list = None, code_type: str = "ISO3A"
+    ) -> tuple:
+        """Return the impact function id and the region corresponding to a list of countries,
+        or a single country.
+
+        Parameters:
+        -----------
+        countries : list
+            List containing the ISO codes of the country, which should be either
+            in string format if the code is "ISO3A" or an integer if "ISO3N", see code_type below.
+        code_type : str
+            Either "ISO3A" or "ISO3N". "ISO3A" stands for "ISO 3166-1 alpha-3" which is a
+            three-letter country code, "ISO3N" stands for "ISO 3166-1 numeric" which is a
+            three-digit country code, the numeric version of "ISO3A". For example, for Switzerland:
+            the "ISO3A" code is "CHE" and the "ISO3N" is 756.
+
+        Returns:
+        --------
+        impf_ids : list
+            List of impact function ids matching the countries.
+        regions_ids : list
+            List of the region ids. Regions are a container of countries as defined in:
+            https://nhess.copernicus.org/articles/21/393/2021/nhess-21-393-2021.pdf, and implemented
+            in the CountryCode Enum Class. Example: "NA1", "NA2", ...
+        regions_names : list
+            List of the regions names. Example: "Caribbean and Mexico", "USA and Canada", ...
+        """
 
-        return region_name[region], impf_id[region], iso3n[region], iso3a[region]
+        if code_type not in {"ISO3A", "ISO3N"}:
```

Comment:
Why is `code_type` a parameter at all? If any item in the list is a `str`, it must be an alpha-3 code. If it is `int`, it must be a numerical code.